### PR TITLE
Update qemu/fvp/hikey on the generic driver

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

File: default.xml [hunk 1]
```diff
@@ -11,14 +11,13 @@
...
 	<project path="optee_client" name="optee_client.git" />
-	<project path="optee_linuxdriver" name="optee_linuxdriver.git" />
 	<project path="optee_test" name="optee_test.git" />
 
 	<!-- busybox -->
 	<project remote="busybox" path="busybox" name="busybox.git" revision="dbf5a6da6a4295ce26edd1ce34fde567d19afa02" />
 
 	<!-- Linux kernel -->
-	<project remote="linux" path="linux" name="linux.git" revision="refs/tags/v4.1-rc1"/>
+	<project remote="linaro-swg" path="linux" name="linux.git" revision="gendrv"/>
```
Comment (jenswi-linaro) [inline default.xml]: revision should be optee

File: fvp.xml [hunk 2]
```diff
@@ -25,13 +24,13 @@
 	<project remote="arm" path="edk2" name="edk2.git" revision="refs/tags/v2.1-rc0" />
 
 	<!-- Linux kernel -->
-	<project remote="linux" path="linux" name="linux.git" revision="refs/tags/v3.18-rc1"/>
+	<project remote="linaro-swg" path="linux" name="linux.git" revision="gendrv"/>
```
Comment (jenswi-linaro) [inline fvp.xml]: revision should be optee

File: hikey.xml [hunk 3]
```diff
@@ -40,10 +38,7 @@
 	<project remote="savannah" path="grub" name="grub" />
 
 	<!-- Linux kernel -->
-	<project remote="96b-hk" path="linux" name="linux" revision="hikey-mainline-rebase" />
-	<!-- project remote="96boards" path="linux" name="linux" revision="hikey" /-->
-	<!-- project remote="linux" path="linux" name="linux" /-->
-
+	<project remote="linaro-swg" path="linux" name="linux" revision="hikey_gendrv"/>
```
Comment (jenswi-linaro) [inline hikey.xml]: revision should be optee

Comment (jforissier) [inline hikey.xml]: +1, I will create a PR to add the `optee` node to `hi6220-hikey.dts`

**Comment by pascal-brand-st-dev** [comment]
Updated

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by pascal-brand-st-dev** [comment]
QEMU: A single test is failing: 7633 (GP specific)

**Comment by jforissier** [comment]
FYI it is also 7633 that was failing on HiKey before I increased the shared memory size.

**Comment by jenswi-linaro** [comment]
It looks like we have two choices, either increase shared memory size for all platforms we intend to do GP testing with or change the shared memory allocator in the driver to use best fit instead of first fit. I guess the shared memory allocator in the old driver used best fit.

Changing to best fit is only a few lines of code in the driver so it's an easy fix.

However, I'm not sure it has to stop merging this.

**Comment by pascal-brand-st-dev** [comment]
Thanks @jforissier. With https://github.com/OP-TEE/optee_os/pull/720,

    Tested-by: Pascal Brand <pascal.brand@linaro.org> (QEMU)
    Tested-by: Pascal Brand <pascal.brand@linaro.org> (QEMU - with Pager)

So either we merge PR 720, either we update the linux driver (but then it requires to update the patches proposed to Linux) and revert https://github.com/OP-TEE/optee_os/pull/709. Either way is fine with me.

Any way, this manifest is OK regarding QEMU.

**Comment by jforissier** [comment]
@jenswi-linaro it's best fit already AFAICT: https://github.com/linaro-swg/linux/blob/optee/drivers/tee/tee_shm_pool.c#L74

**Comment by jenswi-linaro** [comment]
@jforissier yes I noticed that too. :-)

**Comment by vchong** [comment]
@pascal-brand-st-dev  Hi, for hikey.mk, can you also please change the remote for arm-trusted-firmware from 96boards to 96boards-hikey?

**Comment by pascal-brand-st-dev** [comment]
@vchong updated

**Comment by vchong** [comment]
@pascal-brand-st-dev Thanks!

File: fvp.xml [hunk 4]
```diff
@@ -25,13 +24,13 @@
 	<project remote="arm" path="edk2" name="edk2.git" revision="refs/tags/v2.1-rc0" />
```
Comment (jenswi-linaro) [inline fvp.xml]: In the gendrv branch we're using:
`<project remote="landing-teams" path="edk2" name="working/arm/edk2.git" revision="16.01" />`

I just tried it here and it seems to work.

File: fvp.xml [hunk 5]
```diff
@@ -11,7 +11,6 @@
 	<!-- OP-TEE gits -->
 	<project path="optee_os" name="optee_os.git" />
 	<project path="optee_client" name="optee_client.git" />
-	<project path="optee_linuxdriver" name="optee_linuxdriver.git" />
 	<project path="optee_test" name="optee_test.git" />
```
Comment (jenswi-linaro) [inline fvp.xml]: Please add:
`<remote name="landing-teams" fetch="https://git.linaro.org/landing-teams" />`

**Comment by pascal-brand-st-dev** [comment]
updated

**Comment by jenswi-linaro** [comment]
With https://github.com/OP-TEE/build/pull/55
`Tested-by: Jens Wiklander <jens.wiklander@linaro.org> (FVP)`

**Pull request merged**